### DISCUSSION TITLE
希望字体和表情能会退到系统字体，希望支持彩色表情或能会退单色

### DISCUSSION THREAD

**Discussion by yigekuyou** [xfangfang/wiliwili#343] [General]
自带的字体和表情覆盖率不高，自定义字体很困惑
(到底自定义字体要求字体有什么特性🙃)
搜索界面打表情会豆腐，Noto Emoji 支持单色，替换emoji.ttf后并未被使用。
字库补全字体经常为多份字体存在(是这样的)
多语言支持为不同字体文件(adobe-sourcehansans-cn-fonts 和adobe-sourcehanserif-kr-fonts)，
希望本地化能和Linux的本地化目录更兼容 
/usr/share/locale 和locale/*/LC_MESSAGES

**Comment by yigekuyou** [comment]
用noto 替换，发现哪个都不行🙃🙃

**Reply by xfangfang** [reply]
1. 彩色表情暂时无法支持: 如果你可以让 nanovg支持渲染彩色表情，那么wiliwili就可以：https://github.com/memononen/nanovg
2. 自定义字体，我自己测试没什么限制，这里有提供几个可以设置的字体：https://github.com/xfangfang/wiliwili/discussions/38
3. 实测搜索页面打表情不会“豆腐”
<img width="640" alt="image" src="https://github.com/xfangfang/wiliwili/assets/16631584/bb2970f5-6ae6-4699-9713-eb6b43439c61">

4. 希望本地化能和Linux的本地化目录更兼容：因为 wiliwili 支持太多平台了，为这些平台挨个仔细调节对开发者来说有点困难，而且对于一个绝大部分用户都是使用中文的软件，我感觉在这上面花太多时间性价比不高，不知道放在本地化目录对应用会有哪些优势呢？（单纯一问，按现在的设计很难很优雅的在linux将多语言文件存放在不同目录，所以大概率不会修改）
5. 字库补全字体经常为多份字体存在(是这样的)：没看懂
6. Noto Emoji 支持单色，替换emoji.ttf后并未被使用：默认就是 Noto Emoji 
7. 多语言支持为不同字体文件(adobe-sourcehansans-cn-fonts 和adobe-sourcehanserif-kr-fonts)：欢迎提交PR

**Reply by yigekuyou** [reply]
本地化目录是指的locale目录的使用对应的 resources/i18n 这种语言文件的位置
优势就是文件很大时能单独创建语言包 ，其实也没啥用

字库补全字体是指天珩字库这种大型补全 它拥有四个字体文件

**Reply by xfangfang** [reply]
所有和字体相关的修改都可以去看 nanovg 这个库，能让他支持，wiliwili 就能支持
